File: lib/models/reports.dart
class Reports {
  final int id;
  final int userId;
  final int? fbcId;
  final int? liverId;
  final int? urineId;
  final int? fbsId;
  final int? lipidId;
  final int? bpId;
  final DateTime reportDate;
  final String summary;
  final String? imageUrl;

  Reports({
    required this.id,
    required this.userId,
    this.fbcId,
    this.liverId,
    this.urineId,
    this.fbsId,
    this.lipidId,
    this.bpId,
    required this.reportDate,
    required this.summary,
    this.imageUrl,
  });

  // Factory constructor to create a Reports instance from a map (JSON)
  factory Reports.fromJson(Map<String, dynamic> json) {
    return Reports(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      fbcId: json['fbc_id'] as int?,
      liverId: json['liver_id'] as int?,
      urineId: json['urine_id'] as int?,
      fbsId: json['fbs_id'] as int?,
      lipidId: json['lipid_id'] as int?,
      bpId: json['bp_id'] as int?,
      reportDate: DateTime.parse(json['report_date'] as String),
      summary: json['summary'] as String,
      imageUrl: json['image_url'] as String?,
    );
  }

  // Method to convert a Reports instance to a map (JSON)
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'fbc_id': fbcId,
      'liver_id': liverId,
      'urine_id': urineId,
      'fbs_id': fbsId,
      'lipid_id': lipidId,
      'bp_id': bpId,
      'report_date': reportDate.toIso8601String(),
      'summary': summary,
      'image_url': imageUrl,
    };
  }
}

File: models-kotlin/Reports.kt
package com.healthtracker.models

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

data class Reports(
    val id: Int,
    @SerializedName("user_id")
    val userId: Int,
    @SerializedName("fbc_id")
    val fbcId: Int? = null,
    @SerializedName("liver_id")
    val liverId: Int? = null,
    @SerializedName("urine_id")
    val urineId: Int? = null,
    @SerializedName("fbs_id")
    val fbsId: Int? = null,
    @SerializedName("lipid_id")
    val lipidId: Int? = null,
    @SerializedName("bp_id")
    val bpId: Int? = null,
    @SerializedName("report_date")
    val reportDate: LocalDateTime,
    val summary: String,
    @SerializedName("image_url")
    val imageUrl: String? = null
) {
    companion object {
        private val dateFormatter = DateTimeFormatter.ISO_LOCAL_DATE_TIME

        fun fromJson(json: Map<String, Any?>): Reports {
            return Reports(
                id = json["id"] as Int,
                userId = json["user_id"] as Int,
                fbcId = json["fbc_id"] as Int?,
                liverId = json["liver_id"] as Int?,
                urineId = json["urine_id"] as Int?,
                fbsId = json["fbs_id"] as Int?,
                lipidId = json["lipid_id"] as Int?,
                bpId = json["bp_id"] as Int?,
                reportDate = LocalDateTime.parse(json["report_date"] as String, dateFormatter),
                summary = json["summary"] as String,
                imageUrl = json["image_url"] as String?
            )
        }
    }

    fun toJson(): Map<String, Any?> {
        return mapOf(
            "id" to id,
            "user_id" to userId,
            "fbc_id" to fbcId,
            "liver_id" to liverId,
            "urine_id" to urineId,
            "fbs_id" to fbsId,
            "lipid_id" to lipidId,
            "bp_id" to bpId,
            "report_date" to reportDate.format(dateFormatter),
            "summary" to summary,
            "image_url" to imageUrl
        )
    }
}